backend/src/akvo/lumen/auth.clj

```diff
@@ -81,3 +84,65 @@
 (s/def ::keycloak ::keycloak/data)
 (defmethod ig/pre-init-spec :akvo.lumen.auth/wrap-jwt [_]
   (s/keys :req-un [::keycloak]))
+
+(hugsql/def-db-fns "akvo/lumen/lib/dataset.sql")
+
+(defn wrap-auth-datasets
+  "Add to the request the auth-datasets allowed to the current authenticated user,
+  using flow-api check_permissions"
+  [tenant-manager flow-api]
+  (fn [handler]
+    (let [auth-calls #{(with-meta ["/api/library"] {:methods #{:get}})
```

Comment:
I guess this will be moved to the endpoints themselves?

Reply:
I guess too 👍